Firestore/Example/Tests/Integration/API/FIRQueryTests.m

```diff
@@ -211,4 +212,43 @@ - (void)testCanQueryByDocumentIDsUsingRefs {
   XCTAssertEqualObjects(FIRQuerySnapshotGetData(docs), (@[ testDocs[@"ab"], testDocs[@"ba"] ]));
 }
 
+- (void)testQueriesFireFromCacheWhenOffline {
+  NSDictionary *testDocs = @{
+    @"a" : @{@"foo" : @1},
+  };
+  FIRCollectionReference *collection = [self collectionRefWithDocuments:testDocs];
+
+  FIRQueryListenOptions *options = [[[FIRQueryListenOptions options]
+      includeDocumentMetadataChanges:YES] includeQueryMetadataChanges:YES];
+  id<FIRListenerRegistration> registration =
+      [collection addSnapshotListenerWithOptions:options
+                                        listener:self.eventAccumulator.valueEventHandler];
+
+  FIRQuerySnapshot *querySnap = [self.eventAccumulator awaitEventWithName:@"initial event"];
+  XCTAssertEqualObjects(FIRQuerySnapshotGetData(querySnap), @[ @{ @"foo" : @1 } ]);
+  XCTAssertEqual(querySnap.metadata.isFromCache, NO);
+  XCTestExpectation *networkDisabled = [self expectationWithDescription:@"disable network"];
+  [collection.firestore.client disableNetworkWithCompletion:^(NSError *error) {
```

Comment:
Can you use the spanky new `eventAccumulator.errorEventHandler` here?

Reply:
Hrm...  I'm not sure I'm a fan of this spanky new errorEventHandler. I'm leaving this as-is, but will send a follow-up PR to you with an alternative.